src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
```

Comment:
PullProofAI

There is a potential issue with handling errors when decoding the JWT token. It's advisable to add error handling to prevent the application from crashing if there are issues with decoding the token.
```diff
+     try {
+         const userDetail = this.jwt.decode(token) || {};
+     } catch (error) {
+         // Handle error, e.g., log it or set a default value
+         this.logger.error(`Error decoding JWT token: ${error.message}`);
+         const userDetail = {};
+     }
```



<!-- This is an auto-generated comment by PullProofAI -->